onnxscript/converter.py

```diff
@@ -481,10 +496,17 @@ def translate_attr(
         * Expr must be an attribute-reference, that is a name representing an
         attribute-parameter of a containing function.
         """
+
         if isinstance(expr, ast.Name):
             val = self.lookup(expr.id, self.source_of(expr))
             if isinstance(val, values.AttrRef):
-                return self.ir_builder.make_attr_ref(attr_name, val.value, val.typeinfo)
+                attr_ref = self.ir_builder.make_attr_ref(attr_name, val.value, val.typeinfo)
+                if attr_meta and (attr_ref.type != attr_meta.type):
+                    self.fail(
+                        expr,
+                        f"Attribute type {attr_ref.type} does not match expected type {attr_meta.type}",
```

Comment:
Quote user inputs for clarity?

```suggestion
                        f"Attribute type '{attr_ref.type} 'does not match expected type '{attr_meta.type}'",
```